### PR TITLE
Fix ovh_domain_zone_record with missing record id

### PR DESCRIPTION
**Pull request by nledez**
Fix issue #214

### CODE REVIEW

**Comment by jgourmelen** [comment]
hi,
when this fix will be released ?

File: ovh/resource_domain_zone_record.go [hunk 1]
```diff
@@ -160,6 +160,10 @@ func resourceOvhDomainZoneRecordRead(d *schema.ResourceData, meta interface{}) e
 	record, err := ovhDomainZoneRecord(provider.OVHClient, d, d.Id(), d.IsNewResource())
 
 	if err != nil {
+		if errOvh, ok := err.(*ovh.APIError); ok && errOvh.Code == 404 {
```
Comment (yanndegat) [inline ovh/resource_domain_zone_record.go:163]: normally we use this helper func for this purpose

```
return helpers.CheckDeleted(d, err, endpoint)
```

File: ovh/resource_domain_zone_record.go [hunk 2]
```diff
@@ -220,7 +224,9 @@ func resourceOvhDomainZoneRecordDelete(d *schema.ResourceData, meta interface{})
 	)
 
 	if err != nil {
-		return fmt.Errorf("Error deleting OVH Record: %s", err)
+		if errOvh, ok := err.(*ovh.APIError); ok && errOvh.Code != 404 {
```
Comment (yanndegat) [inline ovh/resource_domain_zone_record.go:227]: idem here
``` 
return helpers.CheckDeleted(d, err, endpoint)    
```

File: ovh/resource_domain_zone_record.go [hunk 3]
```diff
@@ -269,6 +278,9 @@ func ovhDomainZoneRecord(client *ovh.Client, d *schema.ResourceData, id string,
 	})
 
 	if err != nil {
+		if errOvh, ok := err.(*ovh.APIError); ok && errOvh.Code == 404 {
```
Comment (yanndegat) [inline ovh/resource_domain_zone_record.go:281]: i think this is useless as the call to CheckDeleted right after is meant to check this 404.

**Review by yanndegat — COMMENTED** [review]
hi @nledez 

thank you very much for pointing all these improvements.

there are small changes we could add, notably the use of the CheckDeleted helper shall be preferred if you don't mind.

**Comment by yanndegat** [comment]
superseeded by #231 and #232

**Pull request closed without merging**